inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/AnnotationSystemAPIImpl.java

```diff
@@ -0,0 +1,418 @@
+            createErrorMessage(e.getMessage(), aClientMessage.getClientName());
+        }
+
+    }
+
+    @Override
+    public void handleDeleteAnnotation(ClientMessage aClientMessage) throws IOException {
+        try {
+            CasStorageSession.open();
+            MDC.put(Logging.KEY_REPOSITORY_PATH, repositoryProperties.getPath().toString());
+            SourceDocument sourceDocument = documentService
+                .getSourceDocument(aClientMessage.getProject(), aClientMessage.getDocument());
+
+            CAS cas = documentService.readAnnotationCas(sourceDocument,
+                aClientMessage.getUsername());
+            Type type = null;
+            for (Iterator<Feature> it = cas.getTypeSystem().getFeatures(); it.hasNext(); ) {
+
+                Feature s = it.next();
+                if (s.getDomain().getShortName().equals(aClientMessage.getAnnotationType())) {
+                    type = s.getDomain();
+                }
+            }
+
+            TypeAdapter adapter = annotationService
+                .getAdapter(annotationSystemAPIService.getAnnotationLayer(type.getName()));
+
+
+            AnnotationMessage message = new AnnotationMessage();
+            message.setAnnotationOffsetBegin(selectAnnotationByAddr(cas,
+                aClientMessage.getAnnotationAddress()).getBegin());
+            message.setAnnotationAddress(String.valueOf(aClientMessage.getAnnotationAddress()));
+            message.setDelete(true);
+
+            adapter.delete(documentService.getSourceDocument(aClientMessage.getProject(),
+                aClientMessage.getDocument()),
+                aClientMessage.getUsername(), cas,
+                VID.parse(String.valueOf(aClientMessage.getAnnotationAddress())));
+            CasStorageSession.get().close();
+
+
+            annotationProcessAPI.handleSendUpdateAnnotation(message,
+                String.valueOf(aClientMessage.getProject()),
+                String.valueOf(aClientMessage.getDocument()), String.valueOf(message.getAnnotationOffsetBegin()));
+
+        }  catch (Exception e) {
+            e.printStackTrace();
+            createErrorMessage(e.getMessage(), aClientMessage.getClientName());
+        }
+    }
+
+    @Override
+    public CAS getCasForDocument(String aUser, long aProject, long aDocument)
+    {
+        try {
+            CasStorageSession.open();
+            MDC.put(Logging.KEY_REPOSITORY_PATH, repositoryProperties.getPath().toString());
+            SourceDocument sourceDocument = documentService.getSourceDocument(aProject, aDocument);
+
+            CAS cas = documentService.readAnnotationCas(sourceDocument, aUser);
+
+            CasStorageSession.get().close();
+            return cas;
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            CasStorageSession.get().close();
+            return null;
+        }
+    }
+
+    @Override
+    public void updateCAS(String aUser, long aProject, long aDocument, CAS aCas)
```

Comment:
Is this method used anywhere? Normally, you should open a storage session when you get an incoming request and close it when you are done responding - and you must not keep any references to CASes *outside* of an open session!